eva-stats/stats_variant_warehouse.py

```diff
@@ -0,0 +1,181 @@
+                              "set variants_variant_warehouse = {3}" \
+                              "where assembly_accession = '{0}'" \
+                              "and project_accession = '{1}'" \
+                              "and analysis_accession = '{2}'".format(assembly, project, analysis, count)
+        execute_query(metadata_handle, update_counts_query)
+
+
+def get_dates_from_variant_warehouse(assembly, project, analysis_filename, metadata_handle, mongo_handle):
+    database_name = get_database_name_from_assembly(metadata_handle, assembly)
+    logger.info("Database name to query: {0}".format(database_name))
+    logger.info("Getting counts from files collection for {0}, {1}, {2}".format(assembly, project, analysis_filename))
+    files_collection = mongo_handle[database_name]['files_2_0']
+
+    where_clause = []
+    for analysis, filename in analysis_filename.items():
+        where = {'sid': project, 'fid': analysis, 'fname': filename}
+        where_clause.append(where)
+    cursor_files = files_collection.find({"$or": where_clause},
+                                         {"fid": 1, "sid": 1, "date": 1, "_id": 0})
+
+    for dates in cursor_files:
+        analysis = dates['fid']
+        date = dates['date']
+        logger.info("Update counts for {0}, {1}, {2}".format(assembly, project, analysis))
+        update_date_query = "update eva_stats.stats " \
+                            "set date_processed = '{3}'" \
+                            "where assembly_accession = '{0}'" \
+                            "and project_accession = '{1}'" \
+                            "and analysis_accession = '{2}'".format(assembly, project, analysis, date)
+        execute_query(metadata_handle, update_date_query)
+
+
+def get_handles(private_config_xml_file):
+    properties = get_properties_from_xml_file("production", private_config_xml_file)
+    mongo_hosts_and_ports = str(properties['eva.mongo.host'])
+    mongo_host, mongo_port = get_mongo_primary_host_and_port(mongo_hosts_and_ports)
+    mongo_username = str(properties['eva.mongo.user'])
+    mongo_password = str(properties['eva.mongo.passwd'])
+
+    mongo_handle = get_mongo_connection_handle(mongo_username, mongo_password, mongo_host)
+    metadata_handle = psycopg2.connect(get_pg_metadata_uri_for_eva_profile(
+        "development", private_config_xml_file), user="evadev")
+
+    return mongo_handle, metadata_handle
+
+
+@click.option("--private-config-xml-file", help="ex: /path/to/eva-maven-settings.xml", required=True)
+@click.option("--projects", help="PRJEB27233,PRJEB36318", required=False)
+@click.command()
+def get_stats(private_config_xml_file, projects):
+    logger.info("Started stats counts from variant warehouse")
+    mongo_handle, metadata_handle = get_handles(private_config_xml_file)
+    logger.info("Got connection handles to mongo and postgres")
+    get_from_variant_warehouse(mongo_handle, metadata_handle, projects)
```

Comment:
"projects" parameter is not mandated but we rely on it as an argument to the get_from_variant_warehouse function. The function does not seem to handle empty projects array either.

Reply:
May be this is already slated as an enhancement, but why don't we look up **eva_stats.stats** table and compare it against EVAPRO to find the new projects/analyses (in case the project accessions are not supplied in the command line)?

Reply:
Good idea, it was implemented so when no projects are passed it compares data in evapro with eva_stats and run the counts for the projects that don't have counts yet